lib/src/map/flutter_map_state.dart

```diff
@@ -34,17 +32,24 @@ class FlutterMapState extends MapGestureMixin
 
     mapController.state = this;
 
-    // Initialize all variables here, if they need to be updated after the map changes
-    // like center, or bounds they also need to be updated in build.
-    _rotation = options.rotation;
-    _center = options.center;
-    _zoom = options.zoom;
-    _pixelBounds = getPixelBounds(zoom);
-    _bounds = _calculateBounds();
+    LatLng center = options.center;
+    double zoom = options.zoom;
+    if (options.bounds != null) {
+      final target =
+          getBoundsCenterZoom(options.bounds!, options.boundsOptions);
+      center = target.center;
+      zoom = target.zoom;
+    }
+
+    // Initialize core state here. State that needs to be updated after a map
+    // changes like: move center, or zoom in/out.
+    _state = ValueNotifier(_State(options.crs, center, zoom, options.rotation));
 
-    WidgetsBinding.instance.addPostFrameCallback((_) {
-      options.onMapReady?.call();
-    });
+    if (options.onMapReady != null) {
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        options.onMapReady!.call();
```

Comment:
I would use a variable to beneficiate from type shadowing
```dart
final onMapReady = options.onMapReady;
if (onMapReady != null) {
  WidgetsBinding.instance.addPostFrameCallback((_) {
    options.onMapReady.call();
  }
}
```

Reply:
Hi [Guillaume](https://github.com/TesteurManiak), nice to meet you. I adapted the PR to your proposal - thanks.

Quickly taking a step back to be mindful of everyone's time: do you think this PR overall is going in the right direction? Just want to avoid a late surprise, e.g. us finding the change too large/invasive. Either way, happy to work though any issues with you. Cheers.

Reply:
Hi, @ignatz I need myself to take a deeper look into your code which I haven't time to do for now. Your contribution is very much welcome, especially if code responsibility is better handled, but it might need some testing to ensure it doesn't cause any regression.

Reply:
Sounds good. Keep me posted. Thanks